### PR TITLE
fix: save enrollment details to local keystore

### LINKED ISSUE #556: feat: at_onboarding_cli store enrollment details in local secondary after apkam auth is success
### Is your feature request related to a problem? Please describe.

* https://github.com/atsign-foundation/at_client_sdk/issues/1269
Once this feature is merged, in order to enforce apkam authorization checks on client side, we need enrollment details on the local secondary to retrieve namespaces and access

### Describe the solution you'd like

* Store enrollment details (namespace and access) on the client side after apkam auth is success.
* Enroll details are required on local storage for at_client to perform enrollment authorization checks on local secondary verbs.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by murali-shris**
Closes https://github.com/atsign-foundation/at_libraries/issues/556
Enroll details are needed in local storage for at_client to perform enrollment authorization checks for local secondary verbs. While we started implementing  authorization checks in at_client, we wanted to use enroll:fetch instead of local key to get enrollment details. But enroll:fetch approach will not work for 1) offline at_client 2) every verb operation will result in remote call. So we decided to store a local key for enrollment details

**- What I did**
- Store enrollment details on LocalSecondary after approval and successful auth using a new enrollment
- Handle edge cases, when null SelfEncryptionKey or EncryptionPrivateKey is received from server
- Changes to OnboardingService.close() :
- - call AtLookup.close() only if AtLookup.isConnectionAvailable is true (to NOT trigger the null exception)
- - Also close atClient.notificationService as this might keep a monitor connection running in the  background in some cases
- Changed OnboardingService._readAtKeysFile () to visibleForTesting
- introduced a setter for enrollmentBase to allow injection of mocks (only visibleForTesting)
- Formatted method docs
- Some refactoring to support mocking

**- How to verify it**
- unit tests and functional tests added to assert the changes

**- Description for the changelog**

fix: save enrollment details to local keystore

### CODE REVIEW

**Review by sitaram-kalluri — DISMISSED** [review]

**Comment by srieteja** [comment]
waiting for https://github.com/atsign-foundation/at_libraries/pull/606 to be merged into this branch.

**Comment by gkc** [comment]
> waiting for #606 to be merged into this branch.

#606 has been merged to trunk

**Comment by gkc** [comment]
> > waiting for #606 to be merged into this branch.
> 
> #606 has been merged to trunk

@srieteja I'll resolve the merge conflicts and merge trunk into this branch

**Comment by srieteja** [comment]
@gkc once the conflicts are resolved, functional tests will fail as a result of changes. Working on updating them

**Comment by gkc** [comment]
> @gkc once the conflicts are resolved, functional tests will fail as a result of changes. Working on updating them

I've fixed all the ripple effects and pushed to the branch

**Review by gkc — CHANGES_REQUESTED** [review]
1. The PR description explains what has been done, but not _why_. Please add an explanation of the problem that is being addressed here (or link to a ticket which has that explanation).
2. This PR handles writing to the keystore ... is the intent that there will be code elsewhere which reads this information from the keystore?
3. The information is currently written to `'${enrollmentResponse.enrollmentId}.new.enrollments.__manage${atClient!.getCurrentAtSign()}'`
  - Given that this is purely for local client-side use, we should use a local key
  - And we should not be using the __manage namespace

**Comment by murali-shris** [comment]
> 1. The PR description explains what has been done, but not _why_. Please add an explanation of the problem that is being addressed here (or link to a ticket which has that explanation).
> 2. This PR handles writing to the keystore ... is the intent that there will be code elsewhere which reads this information from the keystore?
> 3. The information is currently written to `'${enrollmentResponse.enrollmentId}.new.enrollments.__manage${atClient!.getCurrentAtSign()}'`
> 
> * Given that this is purely for local client-side use, we should use a local key
> * And we should not be using the __manage namespace

1) added to the PR description
2) the local key which stores the enrollment details will be read from LocalSecondary --> isEnrollmentAuthorizedForOperation(..) in at_client
3) agreed. Will make the change

File: tests/at_onboarding_cli_functional_tests/test/enrollment_test.dart [hunk 1]
```diff
@@ -361,13 +376,18 @@ void main() {
         sharedBy: atsign,
       ).build();
       String expectedExceptionMessage =
-          'Exception: Cannot perform update on ${buzzKey} due to insufficient privilege';
+          'Exception: Cannot perform update on $buzzKey due to insufficient privilege';
 
       // The put operation is expected to fail as the new enrollment only has
       // access to wavi namespace
-      expect(() => client?.put(buzzKey, 'value'), throwsA(predicate((e) {
+      expect(() async => await client?.put(buzzKey, 'value'),
```
Comment (gkc) [inline tests/at_onboarding_cli_functional_tests/test/enrollment_test.dart]: it's cleaner to do `await expectLater(...)`

**Review by gkc — CHANGES_REQUESTED** [review]
Please un-skip the test(s) in test/at_onboarding_cli_test.dart

**Review by gkc — APPROVED** [review]
LGTM

**Pull request merged**